### PR TITLE
Use generic type in Select prompts on examples/20_tokio_tradeclient

### PR DESCRIPTION
**Pull request by mikaelmello**
Hey, creator of inquire here :)

I'm taking a look at a lot of crates using inquires in order to find the main points when using the lib so I can have a good idea of where to focus my efforts on API stabilization.

So I took the time to suggest these changes that make use of inquire's ability to receive any type that implements Display in its selection prompts. So for UserAction we can pass the whole object, meanwhile for the generated fix42 enums we can use the newtype pattern to create a new type specific to the 
prompt, which implements Display. This helps us to greatly simplify the prompt construction and avoid a serialization-deserialization cycle, at least on the strum_fix42 enum although the original fix42 definition is not something we can change I suppose.

Hope it helps!

### CODE REVIEW

File: examples/20_tokio_tradeclient/src/main.rs
Comment (mikaelmello) [inline examples/20_tokio_tradeclient/src/main.rs:47]: While I have your attention, based on your very simple usage (no further customization on the prompt), how much would you prefer the current syntax (using the full builder pattern) vs this function call?

```suggestion
    let s = inquire::prompt_selection("Select an action", UserAction::iter().collect())?;
    
    // or perhaps we could also omit inquire here since the fn name is verbose enough
    let s = prompt_selection("Select an action", UserAction::iter().collect())?;
```

Comment (neysofu) [reply examples/20_tokio_tradeclient/src/main.rs:47]: I like this a lot, thank you for the suggestion! I'll apply it.

Comment (neysofu) [reply examples/20_tokio_tradeclient/src/main.rs:47]: Mmm, I'm actually having a hard time finding the `prompt_selection` API you're suggesting. Neither a search on docs.rs nor one on the `inquire` GH repo give any results. I might be missing something here 🤔

**Review by neysofu — APPROVED** [review]
I didn't expect to get a PR from the man himself 😛 thank you for `inquire`, it greatly simplified this example.

We do have control over the original `fix42` code generation and thus the derive's, but I'd like not pollute the original `fix42` module itself with `strum` accessories; I'd rather generate a new module and map all types instead.

I'm approving the PR, however I was unable to apply your `prompt_selection` suggestion – I left a comment there, I'll wait for you to review it before merging.

Thanks again for both `inquire` and the PR!

File: examples/20_tokio_tradeclient/src/main.rs
Comment (mikaelmello) [reply examples/20_tokio_tradeclient/src/main.rs:47]: I was misleading sorry... this function hasn't been released yet, but I'm toying around with the idea of exposing a "one-shot" api for the simplest of use cases. Either something like `prompt_confirmation(msg)`, or maybe a macro where I can still have a small degree of customization `prompt_confirmation!($msg) {} prompt_confirmation!($msg, $default) {}` you know?

Comment (neysofu) [reply examples/20_tokio_tradeclient/src/main.rs:47]: Gotcha! Yeah I'd use `prompt_selection`, but only if a similar option is available for `Text` – I'd prioritize consistency over saving a few keystrokes. As for macros, I tend to avoid them unless absolutely necessary.

**Comment by mikaelmello** [comment]
>We do have control over the original fix42 code generation and thus the derive's, but I'd like not pollute the original fix42 module itself with strum accessories; I'd rather generate a new module and map all types instead.

Yeah I noticed that and it makes a lot of sense.

Your example and others made me realize how menu-like prompts are extremely common, and people are either using `strum` only for that or creating the option array themselves. Due to that, I'll see how feasible it is to implement something in inquire tailored to this use-case.

First, a proc macro to generate the enum variant list + display without the need for strum, e.g.:

```
#[derive(PromptSelectable)]
pub enum UserAction {
  #[prompt_value("Enter order")]
  EnterOrder,
  Quit,
}
```

idk, just blurting out ideas here on a random repo lol. I'm glad you enjoyed inquire!

File: examples/20_tokio_tradeclient/src/main.rs
Comment (mikaelmello) [reply examples/20_tokio_tradeclient/src/main.rs:47]: >As for macros, I tend to avoid them unless absolutely necessary.

It's good to hear that upfront. May I ask why? I don't know a lot about its impacts, or if they are a zero-cost abstraction at all.

**Comment by neysofu** [comment]
Oh that's really cool. I'd think about it in terms of a `clap`-like interface, or even make it entirely `clap`-compatible. Now sure how feasible is it as the semantics may not map exactly 1:1, but the idea of taking a Clap subcommand and instantiating its `struct` entirely via prompts is very attractive.

File: examples/20_tokio_tradeclient/src/main.rs
Comment (neysofu) [reply examples/20_tokio_tradeclient/src/main.rs:47]: They are zero-cost indeed, and extremely useful as you can do lots of things you normally wouldn't be able to do. I'm using quite a few in FerrumFIX alone so I'm not against them in principle.

There are two major drawbacks, the way I see it, and they are quite related with each other. (1) The error messages you get with them are quite cryptic and (2) they degrade IDE development experience, as you lose autocomplete, type hints, and other goodies. Also, debugging proc macro -generated code can be very painful, although that almost only concerns the crate author and not end users. For these reasons, I tend to limit proc macro usage to situations where alternatives are way too verbose or outright don't work. Declarative macros somewhat share these problems but they tend to be nicer to work with, in my experience.

Comment (neysofu) [reply examples/20_tokio_tradeclient/src/main.rs:47]: Oh, I forgot: they also quite increase compilation times.

**Pull request merged**
